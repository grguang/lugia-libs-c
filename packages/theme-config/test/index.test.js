// @flow

import * as React from 'react';
import Theme from '../src';

import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

describe('Theme', () => {
  it('Theme', () => {
    class TestComponent extends React.Component<any> {
      render() {
        return (
          <div>
            {JSON.stringify(this.context.config)}
            {this.props.children}
          </div>
        );
      }
    }

    TestComponent.contextTypes = {
      config: PropTypes.object,
    };
    const config = { ligx: { value: '正念' } };
    const cmp = renderer.create(
      <Theme config={config}>
        <TestComponent>hello everyone</TestComponent>
      </Theme>,
    );
    expect(cmp).toMatchSnapshot();
  });
});
