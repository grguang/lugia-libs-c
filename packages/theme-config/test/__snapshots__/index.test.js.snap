// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Theme 1`] = `
<div>
  {"ligx":{"value":"正念"}}
  hello everyone
</div>
`;
